langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java

```diff
@@ -419,66 +382,153 @@ public void callingLengthModificationFunctionsOnFixedLengthLists() {
+                "invalid member type of the array to sort: type '(string|int)[]' is not an ordered type",
+                136, 33);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "invalid member type of the array to sort: type '(string|int)[]' is not an ordered type",
+                138, 33);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "invalid member type of the array to sort: type '(string|int)[]' is not an ordered type",
+                140, 33);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "invalid array sort key function return type: type '(string|int)' is not an ordered type",
+                142, 33);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "invalid member type of the array to sort: type 'map<string>?[]' is not an ordered type",
+                148, 35);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "incompatible types: expected 'function ((any|error)) returns ((boolean|int|float|decimal|string|" +
+                        "(boolean|int|float|decimal|string)?[])?)?', found 'function (map<string>?) " +
+                        "returns (map<string>?)'",
+                150, 62);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "too many arguments in call to 'sort()'", 154, 24);
         Assert.assertEquals(negativeResult.getErrorCount(), errorIndex);
     }
 
     @Test
     public void testShiftOperation() {
         BRunUtil.invoke(compileResult, "testShiftOperation");
     }
+
+    @Test
+    public void testSort1() {
+        BRunUtil.invoke(compileResult, "testSort1");
+    }
+
+    @Test
+    public void testSort2() {
+        BRunUtil.invoke(compileResult, "testSort2");
+    }
+
+    @Test
+    public void testSort3() {
+        BValue[] returns = BRunUtil.invoke(compileResult, "testSort3");
+
+        assertEquals(returns[0].getType().getTag(), TypeTags.ARRAY_TAG);
+        BValueArray arr = (BValueArray) returns[0];
+
+        assertEquals(arr.elementType.getTag(), TypeTags.INT_TAG);
+        assertEquals(arr.size(), 100);
+
+        for (int i = 1; i < arr.size(); i++) {
+            assertTrue(arr.getInt(i) < arr.getInt(i - 1));
+        }
+    }
+
+    @Test
+    public void testSort4() {
+        BRunUtil.invoke(compileResult, "testSort4");
+    }
+
+    @Test
+    public void testSort5() {
+        BRunUtil.invoke(compileResult, "testSort5");
+    }
+
+    @Test
+    public void testSort6() {
+        BRunUtil.invoke(compileResult, "testSort6");
+    }
+
+    @Test
+    public void testSort7() {
+        BRunUtil.invoke(compileResult, "testSort7");
+    }
+
+    @Test
+    public void testSort8() {
+        BRunUtil.invoke(compileResult, "testSort8");
+    }
+
+    @Test
+    public void testSort9() {
+        BRunUtil.invoke(compileResult, "testSort9");
+    }
```

Comment:
Shall we add a couple of tests for the `array:sort()` style of usage as well? Both positive and negative.

Reply:
And what about arrays having finite types?

Reply:
> Shall we add a couple of tests for the `array:sort()` style of usage as well? Both positive and negative.

Have already added for both.
Some positive cases have been added in
[#L701](https://github.com/ballerina-platform/ballerina-lang/blob/62d470fa9d5069bfa89d83bc91d049141beb1712/langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal#L701)
[#L793](https://github.com/ballerina-platform/ballerina-lang/blob/62d470fa9d5069bfa89d83bc91d049141beb1712/langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal#L793)
[#L853](https://github.com/ballerina-platform/ballerina-lang/blob/62d470fa9d5069bfa89d83bc91d049141beb1712/langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal#L853)
Some negative cases have been added in
[#[L136]](https://github.com/ballerina-platform/ballerina-lang/blob/62d470fa9d5069bfa89d83bc91d049141beb1712/langlib/langlib-test/src/test/resources/test-src/arraylib_test_negative.bal#L136)
[#[L148]](https://github.com/ballerina-platform/ballerina-lang/blob/62d470fa9d5069bfa89d83bc91d049141beb1712/langlib/langlib-test/src/test/resources/test-src/arraylib_test_negative.bal#L148)

Reply:
No, I meant using it as a normal function. For example,
```ballerina
int[] sorted = array:sort(arr);
```